cli/azd/pkg/commands/pipeline/pipeline_manager.go

```diff
@@ -200,26 +203,31 @@ func (manager *PipelineManager) Configure(ctx context.Context) error {
 		// changed from "az-cli" to "az-dev"
 		manager.PipelineServicePrincipalName = fmt.Sprintf("az-dev-%s", time.Now().UTC().Format("01-02-2006-15-04-05"))
 	}
+
 	inputConsole.Message(ctx, fmt.Sprintf("Creating or updating service principal %s.\n", manager.PipelineServicePrincipalName))
-	credentials, err := azCli.CreateOrUpdateServicePrincipal(
-		ctx,
-		manager.Environment.GetSubscriptionId(),
-		manager.PipelineServicePrincipalName,
-		manager.PipelineRoleName)
-	if err != nil {
-		return fmt.Errorf("failed to create or update service principal: %w", err)
+	var credentials json.RawMessage = nil
+	var err error
+
+	existingSP := os.Getenv("AZURE_CREDENTIALS")
```

Comment:
What is this environment variable? This seems a little out-of-place. Is this tied to a specific tool or CI we're using? Can we refactor this code to be closer, or document it?

Reply:
I think this was some strategy for re-using an existing service principal, instead of creating a new one (or rotating keys from an existing one).
We don't want to document this yet.  I will remove it for now, and make it part of another PR which describes the entire feature